File: moviper-recyclerview/src/main/java/com/mateuszkoslacz/moviper/iface/view/ViperDataView.java
package com.mateuszkoslacz.moviper.iface.view;

import com.hannesdorfmann.mosby.mvp.MvpView;

/**
 * Created by mateuszkoslacz on 01.12.2016.
 * <p>
 * The root view interface for every VIPER ViewHolder.
 * It shall not be used in Contract.
 */

public interface ViperDataView<DataObject> extends MvpDataView<DataObject>, ViperView, ContextHolder {

}

File: moviper3/src/main/java/com/mateuszkoslacz/moviper3/iface/view/ViperDataView.kt
package com.mateuszkoslacz.moviper3.iface.view

import com.hannesdorfmann.mosby3.mvp.MvpView

/**
 * Created by mateuszkoslacz on 01.12.2016.
 *
 *
 * The root view interface for every VIPER ViewHolder.
 * It shall not be used in Contract.
 */

interface ViperDataView<DataObject> : MvpDataView<DataObject>, ViperView, ContextHolder


interface MvpDataView<DataObject> : MvpView {

    var dataObject: DataObject
}

